components/servo/webview.rs

```diff
@@ -23,6 +23,35 @@ use crate::clipboard_delegate::{ClipboardDelegate, DefaultClipboardDelegate};
 use crate::webview_delegate::{DefaultWebViewDelegate, WebViewDelegate};
 use crate::ConstellationProxy;
 
+/// A handle to a Servo webview. If you clone this handle, it does not create a new webview,
+/// but instead creates a new handle to the webview. Once the last handle is dropped, Servo
+/// considers that the webview has closed and will clean up all associated resources related
+/// to this webview.
+///
+/// ## Rendering Model
+///
+/// Every [`WebView`] has a [`RenderingContext`](crate::RenderingContext). The embedder manages when
+/// the contents of the [`WebView`] paint to the [`RenderingContext`](crate::RenderingContext). When
+/// a [`WebView`] needs to be painted, for instance, because its contents have changed, Servo will
+/// call [`WebViewDelegate::notify_new_frame_ready`] in order to signal that it is time to repaint
+/// the [`WebView`] using [`WebView::paint`].
+///
+/// An example of how this flow might work is:
+///
+/// 1. [`WebViewDelegate::notify_new_frame_ready`] is called. The applications triggers a request
+///    to repaint the window that contains this [`WebView`].
+/// 2. During window repainting, the application calls [`WebView::paint`] and the contents of the
+///    [`crate::RenderingContext`][crate::RenderingContext] are updated.
+/// 3. If the [`crate::RenderingContext`][crate::RenderingContext] is double-buffered, the
+///    application then calls [`crate::RenderingContext::present`] in order to swap the back buffer
+///    to the front, finally displaying the updated [`WebView`] contents.
+///
+/// In cases where the [`WebView`] contents have not been updated, but a repaint is necessary, for
+/// instance when repainting a window due to damage, an application may simply perform the final two
+/// steps and Servo will repaint even without first calling the
+/// [`WebViewDelegate::notify_new_frame_ready`] method.
+///
+/// [`RenderingContext`]: crate::RenderingContext
```

Comment:
Is this needed? in lines 33 and 34 we directly embed the link to the type, so it seems we don't use this referene link?

Reply:
Nope. Removed.